### PR TITLE
Feature/project merge request approval rules

### PR DESCRIPTION
**Pull request by robinson96**
Addresses #1199.

### CODE REVIEW

**Comment by max-wittig** [comment]
Hi @robinson96 thanks for the MR! Would you mind to add some docs and tests to this? I can't really test EE features, but I will take a look based on your docs!

**Comment by robinson96** [comment]
I will certainly add docs. I too lack a great way to do EE testing, I develop [GRAPE](https://github.com/LLNL/GRAPE),  which has some features for maintaining pull requests. It works in that context. (branch unfortunately is on a repo hosting service that is not public).

**Comment by robinson96** [comment]
> 
> 
> Hi @robinson96 thanks for the MR! Would you mind to add some docs and tests to this? I can't really test EE features, but I will take a look based on your docs!

I will certainly add docs. I too lack a great way to do EE testing, I develop [GRAPE](https://github.com/LLNL/GRAPE),  which has some features for maintaining pull requests. It works in that context. (branch unfortunately is on a repo hosting service that is not public).

**Comment by robinson96** [comment]
@max-wittig , I'm trying to find [test_merge_requests.py](https://travis-ci.org/github/python-gitlab/python-gitlab/jobs/735540249#L266) so I can modify it to add appropriate tests, but am coming up short. (My python testing skills are still developing). Any hints on where to be looking? (tools/functional/api/ only has a few of the files that Travis appears to be exercising...)

**Comment by max-wittig** [comment]
Our unittests are here: https://github.com/python-gitlab/python-gitlab/tree/master/gitlab/tests/objects

And functional tests are here: https://github.com/python-gitlab/python-gitlab/tree/master/tools/functional/api

For this, it's only possible to add unit tests, as we run the functional tests against GitLab Core

**Comment by robinson96** [comment]
> 
> 
> Our unittests are here: https://github.com/python-gitlab/python-gitlab/tree/master/gitlab/tests/objects
> 
> And functional tests are here: https://github.com/python-gitlab/python-gitlab/tree/master/tools/functional/api
> 
> For this, it's only possible to add unit tests, as we run the functional tests against GitLab Core

Thanks. Tests are added!

**Comment by robinson96** [comment]
> 
> 
> > Our unittests are here: https://github.com/python-gitlab/python-gitlab/tree/master/gitlab/tests/objects
> > And functional tests are here: https://github.com/python-gitlab/python-gitlab/tree/master/tools/functional/api
> > For this, it's only possible to add unit tests, as we run the functional tests against GitLab Core
> 
> Thanks. Tests are added!

@max-wittig - any other concerns with this PR?

**Review by max-wittig — CHANGES_REQUESTED** [review]
Ah yes sorry I forgot to press the submit review button

File: docs/gl_objects/mr_approvals.rst [hunk 1]
```diff
@@ -50,10 +53,35 @@ Change project-level or MR-level MR approvals settings::
...
+
+Create a MR-level MR approval rule::
+
+	data = {
+	     "name": "my MR custom approval rule",
+	     "approvals_required": 2,
+	     "rule_type": "regular",
+	     "user_ids": [105],
+	     "group_ids": [653, 654],
+	       }
```
[the remark this answers is not in the document]
Comment (robinson96) [reply docs/gl_objects/mr_approvals.rst]: @max-wittig  your edits have been pushed

**Comment by robinson96** [comment]
@max-wittig - any other concerns?

**Comment by robinson96** [comment]
@max-wittig - any other concerns with this branch?

**Comment by max-wittig** [comment]
thanks for the PR and sorry for the time it took to merge

**Comment by robinson96** [comment]
@max-wittig - no worries and thanks for walking me through it!

**Pull request merged**